public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -172,6 +172,22 @@ export function addLabelToQuery(
     )
   );
 
+  // assert that every stream selector doesn't already have this added
+  const everyStreamSelectorAlreadyHasThisAdded = streamSelectorPositions.every((matcherPosition) => {
+    return hasStreamSelectorMatchers.some((matcher) => {
+      // Ignore quotes to check if value has already been added
+      const regex = /`|"|\\/g;
+      const substring = query.substring(matcher.from, matcher.to).replace(regex, '');
+      const valueEscaped = value.replace(regex, '');
```

Comment:
To prevent dealing with quotes in general, just.

```suggestion
      const valueEscaped = value.substring(1, value.length-1);
```

You will find similar code in other parts of the data source.